File: ExtractXML.py
# -*- coding: UTF-8 -*-
#-------------------------------------------------------------------------------
# Name:        ExtractXML.py
#
# Purpose:     Reads a PDF file line by line until "start" text is found.
#              Prints each subsequent line until the "stop" text is found.
#
# Author:      mthornton, based on ExtractXML.java by satchwinston BAE Systems
#
# Created:     2017 FEB 22
# Update:      2017 MAR 01
# Copyright:   (c) mthornton 2017
#-------------------------------------------------------------------------------

import sys

def extractXML(start, stop, filename):
        try:
            with open(filename) as xmlFile:
                while start not in xmlFile.readline():
                    continue
                extracted = []
                #while stop not in xmlFile.readline():
                #    sys.stdout.write(line)
                #    extracted.append(line)
                while True:
                    line = xmlFile.readline()
                    sys.stdout.write(line)
                    extracted.append(line)
                    if stop in line:
                        break
                return extracted
        except:
            print("file not found")

        sys.stdout.flush()


import argparse

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("inputFile")
    parser.add_argument("outputFile")
    args = parser.parse_args()

    xml = extractXML("USCTbankruptcynotice", "</ebn:EBNBatch", args.inputFile)
    with open(args.outputFile, 'w') as outfile:
        outfile.writelines(xml)

File: bankruptcy/xmlUSCT/ExtractXML.java
/**
 * $Id: ExtractXML.java,v 1.2 2014/06/24 19:31:48 dwinston Exp $
 * 
 * @author satchwinston
 * BAE Systems
 *
 *	Reads a PDF file line by line until reaches a specified "start text". Then prints
 *	each subsequent line until the specifed root element name closing tag is reached.
 */

import java.io.File;
import java.io.FileReader;
import java.io.BufferedReader;
import java.io.IOException;

public class ExtractXML {

	private static String filename;
	
	/**
	 * @param args - filename
	 */
	public static void main(String[] args) {
		if (args.length != 1) 
			System.err.println("Usage: java ExtractXML <fname>");
		else {
			File f = new File(args[0]);
			if (!f.exists())
				System.err.println(args[0] + " does not exist");
			else {
				filename = args[0];
				try {
					extractXML("USCTbankruptcynotice", "ebn:EBNBatch");
				}
				catch (Exception err) {
					System.err.println("main:", err.getMessage());
				}
			}
		}
	} // main

	private static void extractXML(String startText, String rootElementName) throws Exception {
		BufferedReader reader = null;
		try {
			reader = new BufferedReader(new FileReader(filename));
			String line = null;
			while ((line = reader.readLine()) != null) {
				if (line.indexOf(startText) > -1)
					break;
			}
			while ((line = reader.readLine()) != null) {
				System.out.println(line);
				if (line.indexOf("</" + rootElementName) > -1)
					break;
			}
		}
		catch (IOException err) {
			System.err.println(err.getMessage());
		}
		finally {
			if (reader != null) 
				reader.close();
		}
	}
}
